tools/c7n_azure/tests/azure_common.py

```diff
@@ -12,115 +12,306 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import datetime
+import email.utils as eut
+import json
 import os
 import re
 
-from c7n_azure import constants
+import msrest.polling
+from azure_serializer import AzureSerializer
+from c7n_azure import constants, actions
 from c7n_azure.session import Session
 from c7n_azure.utils import ThreadHelper
 from mock import patch
+from msrest.pipeline import ClientRawResponse
+from msrest.serialization import Model
+from msrest.service_client import ServiceClient
 from vcr_unittest import VCRTestCase
 
 from c7n.resources import load_resources
 from c7n.schema import generate
 from c7n.testing import TestUtils
 
-import msrest.polling
-from msrest.serialization import Model
-from msrest.service_client import ServiceClient
-from msrest.pipeline import ClientRawResponse
-
 load_resources()
 
 C7N_SCHEMA = generate()
 DEFAULT_SUBSCRIPTION_ID = 'ea42f556-5106-4743-99b0-c129bfa71a47'
 CUSTOM_SUBSCRIPTION_ID = '00000000-5106-4743-99b0-c129bfa71a47'
-DEFAULT_TENANT_ID = 'ea42f556-5106-4743-99b0-c129bfa71a47'
-# latest VCR recording date that tag tests
-# If tests need to be re-recorded, update to current date
-TEST_DATE = datetime.datetime(2018, 9, 10, 23, 59, 59)
+DEFAULT_USER_OBJECT_ID = '00000000-0000-0000-0000-000000000002'
+DEFAULT_TENANT_ID = '00000000-0000-0000-0000-000000000003'
+DEFAULT_STORAGE_KEY = 'DEC0DEDITtVwMoyAuTz1LioKkC+gB/EpRlQKNIaszQEhVidjWyP1kLW1z+jo'\
+                      '/MGFHKc+t+M20PxoraNCslng9w=='
+
+GRAPH_RESPONSE = {
+    "value": [
+        {
+            "NOTE": "THIS RESPONSE FAKED BY AZURE_COMMON.PY",
+            "odata.type": "Microsoft.DirectoryServices.User",
+            "objectType": "User",
+            "objectId": DEFAULT_USER_OBJECT_ID,
+            "displayName": "John Doe",
+            "mail": "john@doe.com",
+            "refreshTokensValidFromDateTime": "2018-08-22T20:37:43Z",
+            "userPrincipalName": "john@doe.com"
+        }
+    ]
+}
+
+ACTIVITY_LOG_RESPONSE = {
+    "value": [
+        {
+            "caller": "john@doe.com",
+            "id": "/subscriptions/ea42f556-5106-4743-99b0-c129bfa71a47/resourcegroups/"
+                  "TEST_VM/providers/Microsoft.Compute/virtualMachines/cctestvm/events/"
+                  "37bf930a-fbb8-4c8c-9cc7-057cc1805c04/ticks/636923208048336028",
+            "operationName": {
+                "value": "Microsoft.Compute/virtualMachines/write",
+                "localizedValue": "Create or Update Virtual Machine"
+            },
+            "eventTimestamp": "2019-05-01T15:20:04.8336028Z"
+        }
+    ]
+}
 
 
 class AzureVCRBaseTest(VCRTestCase):
 
+    TEST_DATE = None
+
+    FILTERED_HEADERS = ['Authorization',
```

Comment:
So later in the `_response_callback` we do a comparison of the response headers where we set make the header key to be the lowered, but these headers preserve their casing, so we will have a mismatch here.

Reply:
Funny I remember explicitly going to verify they were normalized on both sides when I reviewed that change but somehow I misread the list comprehension I guess :)   In this case those are request headers which we delete anyway, but I will certainly clean it up!